uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/LeaveRequestMailNotification.php

```diff
@@ -0,0 +1,391 @@
+<?php
+
```

Comment:
Sounds like a great idea!

Except I'm not sure IF having one template class for each request type will be the best for now.
- Currently the only two template parameters that is added conditionally is just the sickness reason and required documents.
 All the other template logic to fetch parameters are actually shared by all the request types.
- Since all the request types are in the same table, an attribute peculiar to each request type can be accessed in the template once the leave request object is available in each template.
What do you think?